user/user.controller.js

```diff
@@ -1,29 +1,99 @@
 const { User, validate } = require('./user.model')
+const { Token, resetEmailValidate, confirmResetPassword } = require('./token.model')
+const sendEmail = require('../utils/emails/sendEmail')
 const bcrypt = require('bcrypt')
+const crypto = require('crypto')
 const _ = require('lodash')
 
 exports.register = async (req, res) => {
     const { error } = validate(req.body)
     if (error) return res.status(400).send({ detail: error.details[0].message })
 
-    try {
-        let user = await User.findOne({ email: req.body.email })
-        if (user) return res.status(400).send({'detail': 'User with email already exists.'})
+    let user = await User.findOne({ email: req.body.email })
+    if (user) return res.status(400).send({ 'detail': 'User with email already exists.' })
 
-        requestBody = _.pick(req.body, ['_id', 'firstName', 'lastName', 'email', 'password'])
+    requestBody = _.pick(req.body, ['_id', 'firstName', 'lastName', 'email', 'password'])
 
-        if(requestBody.password !== req.body.confirmPassword) return res.status(400).send({detail: 'Password Mismatch'})
+    if (requestBody.password !== req.body.confirmPassword) return res.status(400).send({ detail: 'Password Mismatch' })
 
-        user = new User(requestBody)
-        const salt = await bcrypt.genSalt(10)
-        user.password = await bcrypt.hash(user.password, salt)
+    user = new User(requestBody)
+    const salt = await bcrypt.genSalt(10)
+    user.password = await bcrypt.hash(user.password, salt)
 
-        await user.save()
-        const data = _.pick(user, ['_id', 'firstName', 'lastName', 'email'])
-        res.send(data)
+    await user.save()
+    const data = _.pick(user, ['_id', 'firstName', 'lastName', 'email'])
```

Comment:
Do you really need lodash? You can make your validation only pick the fields you want, that way `req.body` will be same as a `user` json object.